benchmarks/Ruby/cd.rb

```diff
@@ -403,8 +403,8 @@ def right_rotate(y)
   end
 
   def remove_fixup(x, x_parent)
-    while x != @root && (!x || x.color == :black)
-      if x == x_parent.left
+    while !x.equals(@root) && (!x || x.color == :black)
```

Comment:
`.equal?`

Reply:
Weird that the test run in CI doesn't catch that, is this never executed, or only with more iterations?

Reply:
There's some code not reached in the rotations logic I think.
Though, I thought, I did a search and replaces for `.equals?` hmpf. I made that mistake in other places.